client/src/components/course-view/FileLoadDialog.js

```diff
@@ -0,0 +1,104 @@
+// SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import Button from '@mui/material/Button';
+import FormHelperText from '@mui/material/FormHelperText';
+
+// A Dialog component for uploading a file
+
+const instructions = 'Upload a CSV file with the header "studentNo" and headers matching to the study \
+  attainment tags you wish to add grades for. You can see an example of a CSV file of the correct format below.';
+const exampleText = 'A student with the student number 222222 has gotten 8 points from the attainment \'C3I9A1\' and 7 points from attainment \'C3I9A2\'.';
+
+const FileLoadDialog = ({ handleClose, open }) => {
+  const [file, setFile] = useState(null);
+  const [error, setError] = useState('');
+
+  const getFileName = file => file.split('\\').pop();
+
+  const uploadFile = () => {
+    // TODO: validate csv ?? 
+    // TODO: send file to backend
+    console.log('sent: ' + getFileName(file));
+  };
+
+  return (
+    <Dialog open={open} >
+      <DialogTitle >Add Grades from File</DialogTitle>
+      <DialogContent sx={{ pr: 6 }}>
```

Comment:
The padding here is a bit odd. Because of `pr` the content isn't in the middle, so I would delete the `pr: 6`. I would also delete the bottom padding because there seems to be quite a big of a gap, so I would add `pb: 0`.

Reply:
good point!